Meadowlark-js/packages/meadowlark-authz-server/src/handler/RequestToken.ts

```diff
@@ -169,8 +179,11 @@ export async function requestToken(authorizationRequest: AuthorizationRequest):
       }
 
       if (result.response === 'GET_FAILURE_NOT_EXISTS') {
-        Logger.debug(`${moduleName}.requestToken: ${maskClientSecret(requestTokenBody)} 404`, authorizationRequest.traceId);
-        return { body: '', statusCode: 404 };
+        Logger.debug(
+          `${moduleName}.requestToken: ${maskClientSecret(requestTokenBody)} Client does not exist`,
+          authorizationRequest.traceId,
+        );
+        return { body: '', statusCode: 401 };
```

Comment:
404 is not an appropriate response, because it would mean that the token endpoint doesn't exist. In this case, the authentication attempt is simply rejected because of invalid credentials.